chimney/src/main/scala-2/io/scalaland/chimney/internal/compiletime/ConfigurationsPlatform.scala

```diff
@@ -0,0 +1,169 @@
+        implicit val Flag: Type[FlagHead] = typeImpl.fromUntyped(h)
+        implicit val Tail: Type[FlagTail] = typeImpl.fromUntyped(t)
+
+        if (Flag.typeConstructor =:= implicitConflictResolutionTC) {
+          val preference = Flag.typeArgs.head
+          if (preference =:= Type.PreferTotalTransformer) {
+            extractTransformerFlags[FlagTail](defaultFlags).setImplicitConflictResolution(
+              Some(dsls.PreferTotalTransformer)
+            )
+          } else if (preference =:= Type.PreferPartialTransformer) {
+            extractTransformerFlags[FlagTail](defaultFlags).setImplicitConflictResolution(
+              Some(dsls.PreferPartialTransformer)
+            )
+          } else {
+            // $COVERAGE-OFF$
+            c.abort(c.enclosingPosition, "Invalid implicit conflict resolution preference type!!")
+            // $COVERAGE-ON$
+          }
+        } else {
+          extractTransformerFlags[FlagTail](defaultFlags).setBoolFlag[FlagHead](value = true)
+        }
+      } else if (flags.typeConstructor =:= disableTC) {
+        val List(h, t) = flags.typeArgs
+        implicit val Flag: Type[FlagHead] = typeImpl.fromUntyped(h)
+        implicit val Tail: Type[FlagTail] = typeImpl.fromUntyped(t)
+
+        if (flags.typeConstructor =:= implicitConflictResolutionTC) {
+          extractTransformerFlags[FlagTail](defaultFlags).setImplicitConflictResolution(None)
+        } else {
+          extractTransformerFlags[FlagTail](defaultFlags).setBoolFlag[FlagHead](value = false)
+        }
+      } else {
+        // $COVERAGE-OFF$
+        c.abort(c.enclosingPosition, "Bad internal transformer flags type shape!")
+        // $COVERAGE-ON$
+      }
+    }
+
+    protected type CfgTail <: internal.TransformerCfg
+    private val emptyT = typeOf[internal.TransformerCfg.Empty]
+    private val fieldConstTC = typeOf[internal.TransformerCfg.FieldConst[?, ?]].typeConstructor
+    private val fieldConstPartialTC = typeOf[internal.TransformerCfg.FieldConstPartial[?, ?]].typeConstructor
+    private val fieldComputedTC = typeOf[internal.TransformerCfg.FieldComputed[?, ?]].typeConstructor
+    private val fieldComputedPartialTC = typeOf[internal.TransformerCfg.FieldComputedPartial[?, ?]].typeConstructor
+    private val fieldRelabelledTC = typeOf[internal.TransformerCfg.FieldRelabelled[?, ?, ?]].typeConstructor
+    private val coproductInstanceTC = typeOf[internal.TransformerCfg.CoproductInstance[?, ?, ?]].typeConstructor
+    private val coproductInstancePartialTC =
+      typeOf[internal.TransformerCfg.CoproductInstancePartial[?, ?, ?]].typeConstructor
+
+    // TODO: adjust for new config type
+    // TODO: this coule be tailrec
+    private def extractTransformerConfig[From: Type, To: Type, Cfg <: internal.TransformerCfg: Type](
+        runtimeDataIdx: Int
+    ): TransformerConfig[From, To] = {
+      /*
+      val cfgTpe = Type[Cfg].dealias
+
+      if (cfgTpe =:= emptyT) {
+        TransformerConfig()
```

Comment:
This needs to be updated to:
 * add right _typed_ `Field` or `CoproductOverride`
 * add untyped coproduct override in `config.legacy`

Reply:
it's about to be handled in a separate PR, as agreed